### PR TITLE
Get album photos list

### PR DESCRIPTION
**Pull request by shernanm**
## Summary

* The endpoint is implemented that allows you to list the photos of purchased albums.
* Their respective tests are carried out.
* Mocks are made for the response of the external service.

## Trello Card

https://trello.com/c/8pkeQz6a

### CODE REVIEW

File: app/controllers/album.js [hunk 1]
```diff
@@ -57,9 +65,22 @@ exports.buyAlbums = (req, res, next) => {
 
 exports.getAlbumsList = (req, res, next) => {
   logger.info(`getAlbumsList method start, request methods: ${req.method}, endpoint: ${req.path}`);
-  return findAllAlbums(req.params.user_id)
+  const attributes = ['id', 'title'];
+  const { user_id } = req.params;
```
Comment (raparicio6) [inline app/controllers/album.js]: camel case

**Review by raparicio6 — CHANGES_REQUESTED** [review]

File: app/controllers/album.js [hunk 2]
```diff
@@ -33,8 +40,9 @@ exports.buyAlbums = (req, res, next) => {
   const qs = {
     id: req.params.id
   };
+  const attributes = ['id', 'title', 'user_id'];
   logger.info(`buyAlbums method start, request methods: ${req.method}, endpoint: ${req.path}, id: ${qs.id}`);
-  findAlbumBy(qs)
+  findAlbumBy(qs, attributes)
```
Comment (molinajulian) [inline app/controllers/album.js]: Add return here

File: app/controllers/album.js [hunk 3]
```diff
@@ -57,9 +65,22 @@ exports.buyAlbums = (req, res, next) => {
...
     .catch(next);
 };
+
+exports.getAlbumPhotosList = (req, res, next) => {
+  const { id } = req.params;
+  logger.info(
+    `getAlbumPhotosList method start, request methods: ${req.method}, endpoint: ${req.path}, id: ${id}`
+  );
+  const user = decodedToken(req.headers.token);
+  return getAlbumPhotos(id, user.id)
```
Comment (molinajulian) [inline app/controllers/album.js]: Pass object

File: app/services/albums.js [hunk 4]
```diff
@@ -31,10 +31,10 @@ exports.getPhotosBy = qs => {
   });
 };
 
-exports.findAlbumBy = option =>
+exports.findAlbumBy = (condition, attributes) =>
```
Comment (molinajulian) [inline app/services/albums.js]: Pass object

File: app/services/albums.js [hunk 5]
```diff
@@ -46,11 +46,22 @@ exports.albumRegister = album =>
     throw errors.databaseError('Error processing request in database.');
   });
 
-exports.findAllAlbums = userId =>
+exports.findAllAlbumsBy = (condition, attributes) =>
```
Comment (molinajulian) [inline app/services/albums.js]: Same

File: app/services/albums.js [hunk 6]
```diff
@@ -46,11 +46,22 @@ exports.albumRegister = album =>
...
-    attributes: ['id', 'title']
+    where: condition,
+    attributes
   }).catch(() => {
-    logger.info(`Error trying to get the user's albums ${userId}`);
+    logger.info('Error trying to get the users albums');
     throw errors.databaseError('Error processing request in database.');
   });
+
+exports.getAlbumPhotos = (id, userId) => {
```
Comment (molinajulian) [inline app/services/albums.js]: Same

File: test/mocks/albums.js [hunk 7]
```diff
@@ -13,3 +13,26 @@ exports.albumMock = () =>
         }
       ])
   );
+
+exports.albumPhotosMock = () =>
+  beforeEach(() =>
+    nock('https://jsonplaceholder.typicode.com')
```
Comment (molinajulian) [inline test/mocks/albums.js]: Use environment var here

File: app/services/albums.js [hunk 8]
```diff
@@ -46,11 +46,22 @@ exports.albumRegister = album =>
...
+  const attributes = ['id'];
+  return exports.findAlbumBy({ id, userId }, attributes).then(foundAlbum => {
+    if (foundAlbum) {
+      const albumId = foundAlbum.dataValues.id;
+      return exports.getPhotosBy({ albumId }).then(albumsApi => albumsApi.map(({ url }) => url));
```
Comment (molinajulian) [inline app/services/albums.js]: `àlbumsApi` ? photos

Comment (molinajulian) [reply app/services/albums.js]: Also, why map url ? Send all attributes for photos

File: app/services/albums.js [hunk 9]
```diff
@@ -46,11 +46,22 @@ exports.albumRegister = album =>
...
+    }
+    return [];
```
Comment (molinajulian) [inline app/services/albums.js]: Move this for bottom, like `if(!albumApi) return []` or similar

**Review by molinajulian — CHANGES_REQUESTED** [review]

File: app/controllers/album.js [hunk 10]
```diff
@@ -33,7 +40,8 @@ exports.buyAlbums = (req, res, next) => {
   logger.info(
     `buyAlbums method start, request methods: ${req.method}, endpoint: ${req.path}, id: ${req.params.id}`
   );
-  return findAlbumBy(req.params)
+  const attributes = ['id', 'title', 'user_id'];
```
Comment (raparicio6) [inline app/controllers/album.js]: pure constants go with UPPER_CASE

File: app/controllers/album.js [hunk 11]
```diff
@@ -56,7 +64,19 @@ exports.buyAlbums = (req, res, next) => {
 
 exports.getAlbumsList = (req, res, next) => {
   logger.info(`getAlbumsList method start, request methods: ${req.method}, endpoint: ${req.path}`);
-  return findAllAlbumsBy(req.params)
+  const attributes = ['id', 'title'];
```
Comment (raparicio6) [inline app/controllers/album.js]: its dont seem pretty to create a variable for this.
put it directyl in the next line without declaring it

File: package.json [hunk 12]
```diff
@@ -64,6 +64,7 @@
     "jsonwebtoken-promisified": "^1.0.3",
     "jwt-simple": "^0.5.1",
     "morgan": "^1.9.1",
+    "nock": "^10.0.6",
```
Comment (raparicio6) [inline package.json]: we talked to use jest :D

**Review by raparicio6 — CHANGES_REQUESTED** [review]

File: package.json (hunk 12 above)
Comment (shernanm) [reply package.json]: I don't know why that library remains, knowing that I have removed it before.

File: app/controllers/album.js [hunk 13]
```diff
@@ -56,7 +63,18 @@ exports.buyAlbums = (req, res, next) => {
...
 };
+
+exports.getAlbumPhotosList = (req, res, next) => {
+  const { id } = req.params;
+  logger.info(
+    `getAlbumPhotosList method start, request methods: ${req.method}, endpoint: ${req.path}, id: ${id}`
+  );
+  const user = decodedToken(req.headers.token);
+  return getAlbumPhotos({ id, userId: user.id })
+    .then(albumPhotos => res.send({ albumPhotos }))
```
Comment (raparicio6) [inline app/controllers/album.js:78]: do you know the difference between doing
`res.send(json)` and `res.send({json})`?

File: app/services/albums.js [hunk 14]
```diff
@@ -45,14 +45,25 @@ exports.findAlbumBy = option =>
...
+    where: condition,
+    attributes
   }).catch(() => {
     logger.info("Error trying to get the user's albums");
-    throw errors.databaseError('Error processing request in database.');
+    throw errors.databaseError(DATABASE_ERROR);
   });
+
+exports.getAlbumPhotos = ({ id, userId }) => {
+  const attributes = ['id'];
```
Comment (raparicio6) [inline app/services/albums.js]: please revise this with the comment I put before

**Review by raparicio6 — CHANGES_REQUESTED** [review]

File: app/controllers/album.js (hunk 13 above)
Comment (shernanm) [reply app/controllers/album.js:78]: `res.send(json)` The answer would be delivering as it comes from the service, in this case an array.
`res.send({json})` The answer would be delivering an object with the **albumPhotos** property and this in turn contains the array that delivers the service.

**Review by raparicio6 — APPROVED** [review]

**Review by molinajulian — APPROVED** [review]

**Pull request merged**